build/prow/kubekins-e2e/Dockerfile

```diff
@@ -22,11 +22,11 @@
 # between these two files will eliminate any confusion on what versions are
 # actually being used.
 
-ARG KUBEKINS_REGISTRY="gke-test-infra"
-ARG KUBEKINS="v20220204-29ea0e1be-1.22"
+FROM gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230309-9a6b1b3121-1.23
 
-FROM gcr.io/${KUBEKINS_REGISTRY}/kubekins-e2e:${KUBEKINS}
-
-ARG KIND_RELEASE="v0.14.0"
-RUN wget -q -O /bin/kind https://kind.sigs.k8s.io/dl/${KIND_RELEASE}/kind-linux-amd64
+# TODO: using a pre-installed kind binary makes it very painful to upgrade the
+# kind version. Update the e2e tests to either:
+# - install kind binary at runtime
+# - consolidate kind usage to Go dependency rather than mixing usage with binary
+RUN wget -q -O /bin/kind https://kind.sigs.k8s.io/dl/v0.14.0/kind-linux-amd64
```

Comment:
It was hard coded in the Makefile so that it could be set as both the image tag and passed in as a build arg. Now that it's not being used in the image tag, I just moved the hard coding into the Dockerfile. As the TODO mentions I think we should eventually update this so that we aren't installing the kind binary into the prow image. It makes upgrading to versions pretty much impossible without breaking CI

Reply:
I'm fine with not hard coding the values in the Makefile. I kinda liked the args here because it called out that they were versions that needed to be updated periodically, but I guess it's not a big deal.

Reply:
I can put the arg back in the makefile, just figured this would be simpler if the var is only used for passing into the docker build